External/Plugins/CodeRefactor/Commands/InlineRename.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Windows.Forms;
+using PluginCore;
+using ScintillaNet;
+
+namespace CodeRefactor.Commands
+{
+    public class InlineRename : IMessageFilter, IDisposable
+    {
+        static InlineRename current;
+
+        ScintillaControl sci;
+        string oldName, newName;
+        int start, end;
+
+        public event Action<string, string> OnRename;
+        public event Action<string> OnUpdate;
+        public event Action OnCancel;
+
+        public static bool CancelExisting()
+        {
+            if (current == null) return false;
+            current.Cancel();
+            return true;
+        }
+
+        public static Rename Start()
+        {
+            if (current == null)
+                return new Rename(true);
+
+            return null;
+        }
+
+        public InlineRename(ScintillaControl control, string original, int position)
+        {
+            if (current != null)
+                current.Cancel();
+
+            sci = control;
+            start = position;
+            oldName = original;
+            end = start + oldName.Length;
+
+            Application.AddMessageFilter(this);
+            sci.TextInserted += Sci_TextInserted;
+            sci.TextDeleted += Sci_TextDeleted;
+            current = this;
+
+            AddHighlight();
+        }
+
+        public void Dispose()
+        {
+            Application.RemoveMessageFilter(this);
+            sci.TextInserted -= Sci_TextInserted;
+            sci.TextDeleted -= Sci_TextDeleted;
+            current = null;
+        }
+
+        void Rename()
+        {
+            Dispose();
+
+            RemoveHighlight();
+            sci.SetSel(start, end);
+            newName = sci.SelText;
+            sci.ReplaceSel(oldName);
+
+            if (OnRename != null) OnRename(oldName, newName);
+        }
+
+        void Cancel()
+        {
+            Dispose();
+            
+            RemoveHighlight();
+            sci.SetSel(start, end);
+            sci.ReplaceSel(oldName);
+
+            if (OnCancel != null) OnCancel();
+        }
+
+        void Update(int position)
+        {
+            AddHighlight();
+
+            if (OnUpdate != null)
+            {
+                sci.SetSel(start, end);
+                string text = sci.SelText;
+                sci.SetSel(position, position);
+                OnUpdate(text);
+            }
+        }
+
+        void AddHighlight()
+        {
+            //sci.AddHighlight(8, 0x00FF00, start, end - start);
+            ////if error: sci.AddHighlight(6, 0xFF0000, start, end - start);
+        }
+
+        void RemoveHighlight()
+        {
+            //sci.RemoveHighlight(start, end - start);
```

Comment:
thanks it really helped
for my case however, adding `sci.RemoveHighlights(0)` at the beginning was the solution